scripts/deploy-and-test-on-instance.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+set -eu
+
+JOB_UPDATE_STAGING="8d452c26-5a39-4162-b3b8-4da5efdf0b76"
+
+# Exec Rundeck Job
+function executeRundeck {
+  INSTANCE=$1
+  JOB=$2
+  APP_SLUG=$3
+  APP_SOURCE=$4
+
+curl -s -X POST \
+     -H "X-Rundeck-Auth-Token: $RUNDECK_TOKEN" \
+     -H "Content-Type: application/json" \
+     -d "{\"argString\":\"-instance $INSTANCE -slug $APP_SLUG -source $APP_SOURCE \"}"  \
+     https://rundeck.cozycloud.cc/api/27/job/$JOB/run
+}
+
+# return status for an executions
+function getRundeckStatus {
+  EXEC_ID=$1
+  local __resultvar=$2
+
+STATUS=$(curl -s -X GET \
+     -H "X-Rundeck-Auth-Token: $RUNDECK_TOKEN" \
+     -H "Content-Type: application/json" https://rundeck.cozycloud.cc/api/27/execution/$EXEC_ID | grep -o -P "(?<=status=').*(?=' )")
+eval $__resultvar="$STATUS"
```

Comment:
Maybe it's because of my limited bash-scripting knowledge, but I think we need to decide if we want to keep using functions in this script, and if so, how we manage our variables.

I don't think we need the functions — `executeRundeck ` is only called in one place and could be inlined. `getRundeckStatus ` is called twice, but as mentioned in another comment, I think we could call it only once. So my preference would be to not use functions at all and inline the curl calls as much as possible.

If for some reason we want to keep the functions around, the variables usage needs to be cleared up:
- the function gets some variables as parameters, but takes others from the global scope (`RUNDECK_TOKEN `) — we should do one or the other, but not both
- we are creating new global variables (`EXEC_ID` and `STATUS`), but we are updating `JOB_STATUS` with the `eval` trick — again, either this function creates global variables, or it doesn't, but doing both is inconsistent.

So if we keep the function, I would suggest we rely only on global variables and try to not create too many.

Reply:
I remove both functions, hence removing a lot of variables. 
I agree it's "better looking' that way.